data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetProtoArray.java

```diff
@@ -64,10 +83,33 @@ public GetProtoArray(final ChainDataProvider chainDataProvider, final JsonProvid
       })
   @Override
   public void handle(final Context ctx) throws Exception {
-    ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    request.header(Header.CACHE_CONTROL, CACHE_NONE);
+
+    final List<Map<String, String>> data =
+        chainDataProvider.getProtoArrayData().stream()
+            .map(this::mapObjectValuesToString)
+            .collect(Collectors.toList());
+    request.respondOk(data);
+  }
 
-    final GetProtoArrayResponse response =
-        new GetProtoArrayResponse(chainDataProvider.getProtoArrayData());
-    ctx.result(jsonProvider.objectToJSON(response));
+  // TODO check this correct approach to making mapping type work
```

Comment:
I'd be super tempted to make ProtoNode::getData return a Map<String,String>... it's only used for this api...

Reply:
I'd agree with that - makes sense to just return what we need to serialise.  Previously returning typed objects worked best but if string makes things easier that's fine.  It's just debug data so doesn't matter if the exact format of the output changes.